lib/http/client.rb

```diff
@@ -154,6 +154,8 @@ def make_request_uri(uri, opts)
       uri.path = "/" if uri.path.empty?
 
       uri
+    rescue Addressable::URI::InvalidURIError => e
+      raise HTTP::Request::InvalidURIError, e.message
```

Comment:
I wonder if we should move this to `HTTP::URI.parse` instead? And probably call it `HTTP::URI::InvalidError`